lib/travis/hub/support/logs.rb

```diff
@@ -1,28 +1,47 @@
 require 'faraday'
 
 module Travis
-  class Logs < Struct.new(:config)
-    def update(id, msg)
-      http.put("/logs/#{id}", msg)
-    end
+  module Hub
+    module Support
+      class Logs < Struct.new(:config)
+        def update(id, msg, clear: false)
+          http.put do |req|
+            req.url "/logs/#{id}"
+            req.params['clear'] = '1' if clear
+            req.headers['Content-Type'] = 'application/octet-stream'
+            req.body = msg
+          end
+        end
 
-    private
+        def append_log_part(id, part, final: false)
+          http.put do |req|
+            req.url "/log-parts/#{id}/last"
+            req.headers['Content-Type'] = 'application/json'
+            req.body = JSON.dump(
+              '@type' => 'log_part',
+              'encoding' => 'base64',
+              'content' => Base64.encode64(part),
+              'final' => final
+            )
+          end
+        end
 
-      def http
-        Faraday.new(url: url) do |c|
-          c.request  :authorization, :token, token
-          c.request  :retry, max: 5, interval: 0.1, backoff_factor: 2
-          c.response :raise_error
-          c.adapter  :net_http
+        private def http
+          @http ||= Faraday.new(url: url) do |c|
+            c.request :authorization, :token, token
+            c.request :retry, max: 5, interval: 0.1, backoff_factor: 2
```

Comment:
I'm taking it back!  The gatekeeper integration can be as simple as `c.response :raise_error`, imho.  Pushing shortly.